natscontext/context.go

```diff
@@ -599,12 +599,55 @@ func numCreds(c *Context) int {
 	return i
 }
 
-func createTree(parent string) error {
-	return os.MkdirAll(ctxDir(parent), 0700)
+func createTree() error {
+	parent, err := ctxDir()
+	if err != nil {
+		return err
+	}
+
+	return os.MkdirAll(parent, 0700)
+}
+
+func natsConfigDir() (string, error) {
+	var (
+		err    error
+		parent string
+	)
+
+	if parent = os.Getenv("NATS_CONFIG_HOME"); parent == "" {
+		if parent, err = parentDir(); err != nil {
```

Comment:
I am not a huge fan of the `if x, err := thing(); err != nil` style if statements, jsm/cli always expand them to multiple lines.

Seems to me `parentDir()` should probably check the `NATS_CONFIG_HOME` variable so it's always doing the right thing wherever it's called? Same as `parentDir()` handles the xdg path for you

Reply:
I initially went down the path of modifying parentDir, but then pulled back to just the context handling... I'm happy to work on the overall NATS_CONFIG_HOME approach, which would put more of the work here.

Reply:
Still not convinced this is a good idea :)

But either way, we also have to set https://github.com/nats-io/natscli/blob/f9c4d8232b7360eb28ad367e80afab2036537d78/internal/util/config.go#L109